runtime/near-vm-logic/src/logic.rs

```diff
@@ -115,17 +113,16 @@ impl<'a> VMLogic<'a> {
         // Overflow should be checked before calling VMLogic.
         let current_account_balance = context.account_balance + context.attached_deposit;
         let current_storage_usage = context.storage_usage;
-        let max_gas_burnt = if context.is_view {
-            config.limit_config.max_gas_burnt_view
-        } else {
-            config.limit_config.max_gas_burnt
+        let max_gas_burnt = match context.view_config {
+            Some(ViewConfig { max_gas_burnt_view }) => max_gas_burnt_view,
+            None => config.limit_config.max_gas_burnt,
         };
         let current_account_locked_balance = context.account_locked_balance;
         let gas_counter = GasCounter::new(
             config.ext_costs.clone(),
             max_gas_burnt,
             context.prepaid_gas,
-            context.is_view,
+            context.view_config.is_some(),
```

Comment:
This can benefit a bit from giving a name to the expresson:

```
let is_view = context.view_config.is_some();
```

I guess, it's even better to define an `is_view` method, and use it here  and in every `if self.context.view_config.is_some()` check.